src/apps/activities/api/activities.py

```diff
@@ -358,3 +360,76 @@ async def __filter_activities(
                     activities.remove(activity)
 
     return activities
+
+
+async def applet_activities_counters_for_subject(
+    applet_id: uuid.UUID,
+    subject_id: uuid.UUID,
+    user: User = Depends(get_current_user),
+    language: str = Depends(get_language),
+    session=Depends(get_session),
+    answer_session=Depends(get_answer_session),
+) -> Response[ActivitiesCounters]:
+    applet_service = AppletService(session, user.id)
+    await applet_service.exist_by_id(applet_id)
+    await CheckAccessService(session, user.id).check_applet_detail_access(applet_id)
+
+    await SubjectsService(session, user.id).exist_by_id(subject_id)
+
+    # Fetch assigned activity or flow IDs for the subject
+    assigned_ids = await ActivityAssignmentService(session).get_assigned_activity_or_flow_ids_for_subject(subject_id)
+
+    # Fetch activities submissions by the subject
+    submitted_activities = await AnswerService(session, user.id, answer_session).get_submissions_by_subject(subject_id)
+
+    # Fetch auto assigned activity and flow IDs by applet ID
+    auto_activity_ids = await ActivityService(session, user.id).get_activity_and_flow_ids_by_applet_id_auto(applet_id)
+
+    respondent_ids = set(assigned_ids.as_respondent + auto_activity_ids)
+    target_ids = set(assigned_ids.as_target + auto_activity_ids)
+    all_assigned_ids = respondent_ids.union(target_ids)
```

Comment:
This isn't causing any issues, but I find it a little hard to follow the code when we're doing a combination of array concatenation and unions. While I understand that:
- `set(assigned_ids.as_respondent + auto_activity_ids)` and
- `set(assigned_ids.as_respondent).union(auto_activity_ids)`

are functionally equivalent, it's more obvious to me when reading the code that we're protecting against duplicate IDs in the second form. Also, Copilot suggests the second version (using `union`) is more direct and can be more efficient.

```suggestion
    respondent_ids = set(assigned_ids.as_respondent).union(auto_activity_ids)
    target_ids = set(assigned_ids.as_target).union(auto_activity_ids)
    all_assigned_ids = respondent_ids.union(target_ids)
```